plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/CekWithTime.hs

```diff
@@ -0,0 +1,14 @@
+module UntypedPlutusCore.Evaluation.Machine.Cek.CekWithTime where
+
+import           UntypedPlutusCore.Evaluation.Machine.Cek.Internal
+
+import           Data.Time.Clock                                   (getCurrentTime)
+import           System.IO.Unsafe                                  (unsafePerformIO)
+
+emitCekWithTime :: GivenCekEmitter s => String -> CekM uni fun s ()
+emitCekWithTime str =
+    let withTime =
+            "[" ++
+            (show $ unsafePerformIO getCurrentTime) ++
+            "]" ++ str
+    in emitCek withTime
```

Comment:
This likely doesn't do what you want. That `String` part suggests that there is no way we could accidentally fully force the argument of `emitCek` as that would require `deepseq`-ing it, which usually does not happen by accident. And indeed, I don't think we do that: `emitCek` just appends a string to the list with logged messages and the builtin application machinery does not bother `deepseq`-ing anything (as that would have horrible performance). And so that `unsafePerformIO` is forced upon inspection of the list of logged messages after the program is fully evaluated, not during evaluation as it should.

Reply:
Thanks! The time is mapped to each item of the list of log msgs. E.g., this is the failed `plutus-tx` test result (from my other PR):
```
traceRepeatedly:                  FAIL
      Test output was different from 'test/TH/traceRepeatedly.plc.golden'. Output of ["diff","-u","test/TH/traceRepeatedly.plc.golden","/build/traceRepeatedly.plc14-164.actual"]:
      --- test/TH/traceRepeatedly.plc.golden    1970-01-01 00:00:01.000000000 +0000
      +++ /build/traceRepeatedly.plc14-164.actual       2021-07-19 19:04:57.176258400 +0000
      @@ -1 +1,3 @@
      -[Adding them up, Making my first int, Making my second int]
      \ No newline at end of file
      +[ [2021-07-19 19:04:57.176820114 UTC]Adding them up
      +, [2021-07-19 19:04:57.176850892 UTC]Making my first int
      +, [2021-07-19 19:04:57.176861612 UTC]Making my second int ]
      \ No newline at end of file
``` 
Is something like this what we want?

Reply:
How is `traceRepeateadly` defined?

I'm not arguing about the output format, I'm saying that `getCurrentTime` is probably invoked **after** the program is evaluated (not during evaluation like it should) because of laziness.

Reply:
I understand, I'm just wondering if adding the time like this is helpful at all. I'm not completely sure what is wanted so clarification is helpful. I'll see if I can add time **within** the item. Here's the definition of `traceRepeatedly`:

```haskell
traceRepeatedly :: CompiledCode Integer
traceRepeatedly = $$(compile
     [||
               let i1 = trace "Making my first int" (1::Integer)
                   i2 = trace "Making my second int" (2::Integer)
                   i3 = trace "Adding them up" (i1 + i2)
              in i3
    ||])
```

Reply:
I will probably need to do everything in IO but that should be fine just for the purpose of profiling.

Reply:
> I'm just wondering if adding the time like this is helpful at all. I'm not completely sure what is wanted so clarification is helpful. 

Ah, I don't really know. Looks fine to me. Let's go with that format for now and if somebody is unhappy about it, we can always change it.

> that should be fine just for the purpose of profiling.

Only if profiling takes inspecting logs into account. Does it?

> I will probably need to do everything in IO

Nah, we don't have `IO` in `CekM` and can't afford adding it. We do have `ST` though and so you should be able to use `CekCarryingM . unsafeIOToST` to lift `getCurrentTime` into `CekM` to make it execute at the right time.

Reply:
>Only if profiling takes inspecting logs into account. Does it?

I don't know - I'll have to check with Michael.

>> I will probably need to do everything in IO

>Nah, we don't have `IO` in `CekM` and can't afford adding it. We do have `ST` though and so you should be able to use `CekCarryingM . unsafeIOToST` to lift `getCurrentTime` into `CekM` to make it execute at the right time.

Sorry I did mean to say the strict monad. Thanks for the advice!

Reply:
Yeah, I think we want to use `unsafeIOToSt`, which should also ensure that it gets sequenced properly and we don't need to worry about laziness messing with us.

Reply:
Format looks fine!